pkg/git/repository.go

```diff
@@ -229,6 +241,39 @@ var clone = func(ctx context.Context, opts *CloneOptions) (*repo, error) {
 	return repo, nil
 }
 
+var createRepo = func(ctx context.Context, opts *CloneOptions) (string, error) {
+	host, orgRepo, _, _, _, _, _ := util.ParseGitUrl(opts.Repo)
+	providerType := opts.Provider
+	if providerType == "" {
+		u, err := url.Parse(host)
+		if err != nil {
+			return "", err
+		}
+
+		providerType = u.Hostname()
+	}
+
+	p, err := NewProvider(&ProviderOptions{
+		Type: providerType,
+		Auth: &opts.Auth,
+		Host: host,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	s := strings.Split(orgRepo, "/")
```

Comment:
I think this is not a good approach for other git providers. For example gitlab repos: https://gitlab.com/foo22/bar/fizz

Reply:
I think (and we should check) that the parse func will return "foo22/bar/fizz" as orgRepo. So I guess the correct split code should be inside the specific provider.